repose-aggregator/core/repose-core/src/main/java/org/openrepose/nodeservice/httpcomponent/HttpComponentRequestProcessor.java

```diff
@@ -85,7 +88,13 @@ public static HttpUriRequest process(HttpServletRequest servletRequest,
         final RequestBuilder requestBuilder = RequestBuilder.create(servletRequest.getMethod())
             .setUri(getUri(servletRequest, target));
         if (!NO_ENTITY_METHODS.contains(servletRequest.getMethod())) {
-            requestBuilder.setEntity(getEntity(servletRequest, chunkedEncoding));
+            byte[] bytes = new byte[8];
+            PushbackInputStream pushbackInputStream = new PushbackInputStream(servletRequest.getInputStream(), 8);
+            int bytesRead = pushbackInputStream.read(bytes, 0, 8);
```

Comment:
Perhaps replace the `8` with `bytes.length`? One less thing to keep in sync.

Reply:
Legit